test/system/build-testimage

```diff
@@ -12,6 +12,76 @@
 # still need a fedora image for that.
 #
 
+if [ "$(uname -s)" = "Linux" ]; then
+    testimg_base=quay.io/libpod/testimage
+    arch_list="amd64 arm64 ppc64le s390x"
+
+    function busyboximg {
+        local repo=$1
+        echo docker.io/$repo/busybox:1.34.1
+    }
+
+    function build_containerfile {
+        # alpine because it's small and light and reliable
+        #    - check for updates @ https://hub.docker.com/_/alpine
+        # busybox-extras provides httpd needed in 500-networking.bats
+        # iproute2 provides JSON output (not in busybox) for 505-networking-pasta.bats
+        # socat offers convenient UDP test termination in 505-networking-pasta.bats
+        cat <<EOF
+ARG REPO=please-override-repo
+FROM docker.io/\${REPO}/alpine:3.16.2
+RUN apk add busybox-extras iproute2 socat
+ADD testimage-id pause /home/podman/
+RUN rm -f /var/cache/apk/*
+EOF
+    }
+
+    # docker.io repo is usually the same name as the desired arch; except
+    # for arm64, where podman needs to have the arch be 'arm64' but the
+    # image lives in 'arm64v8'.
+    function repo_for_arch {
+        local arch=$1
+        if [[ $repo = "arm64" ]]; then
+            echo ${repo}v8
+        else
+            echo ${repo}
+        fi
+    }
+elif [ "$(uname -s)" = "FreeBSD" ]; then
+    testimg_base=quay.io/dougrabson/testimage
+    arch_list="amd64 arm64"
+    busybox=quay.io/dougrabson/busybox:1.36.1
+
+    function date {
+        gdate "$@"
+    }
+    function mktemp {
+        gmktemp "$@"
+    }
+    function touch {
+        gtouch "$@"
+    }
+
+    function busyboximg {
+        echo quay.io/dougrabson/busybox:1.36.1
+    }
+
+    function build_containerfile {
+        # busybox because it's small and has a command set which is close to what
+        # the test suite needs.
+        cat <<EOF
+FROM ${busybox}
+ADD testimage-id pause /home/podman/
+RUN mkdir /usr
+RUN mkdir /srv
+EOF
+    }
+
+    function repo_for_arch {
+        echo dougrabson
```

Comment:
debugging?

Reply:
Not really. On linux, this builds a multi-arch manifest from single-arch images located at e.g. docker.io/amd64/busybox. For freebsd, my busybox image at quay.io/dougrabson/busybox is a multi-arch manifest so we can use the same image name for both amd64 and arm64 in the code below.